src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -16,29 +18,52 @@ import { Logger } from '../../../core/abstractions/logger.service'
 })
 export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
   @Input() public segment: Segment
-  @Input() public videoClipAction: Tv2VideoClipAction | undefined
+  @Input() public videoClipAction?: Tv2VideoClipAction
+
+  public showOnAirBorder: boolean = false
+  public showNextBorder: boolean = false
 
   private readonly fallbackPreviewUrl: string = 'assets/sofie-logo.svg'
   protected media: Media
   private configurationServiceSubscription: Subscription
-  private studioConfiguration: StudioConfiguration | undefined
+  private studioConfiguration?: StudioConfiguration
   private readonly logger: Logger
   protected mediaDurationInMsWithoutPostroll: number = 0
 
   constructor(
     private readonly actionService: ActionService,
     private readonly configurationService: ConfigurationService,
     private readonly mediaStateService: MediaStateService,
+    private readonly rundownStateService: RundownStateService,
     logger: Logger
   ) {
     this.logger = logger.tag('MiniShelfComponent')
   }
 
-  public ngOnInit(): void {
+  public async ngOnInit(): Promise<void> {
     this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((studioConfiguration: StudioConfiguration) => {
       this.studioConfiguration = studioConfiguration
     })
     this.updateMediaAndCalculate()
+
+    this.rundownStateService.subscribeToOnAirPart(this.segment.rundownId).subscribe(onAirPart => {
+      const tv2Part: Tv2Part | undefined = onAirPart as Tv2Part | undefined
+      this.showOnAirBorder = !tv2Part || !this.videoClipAction ? false : tv2Part?.metadata?.actionId === this.videoClipAction?.id
+    })
+
+    this.rundownStateService.subscribeToNextPart(this.segment.rundownId).subscribe(nextPart => {
+      const tv2Part: Tv2Part | undefined = nextPart as Tv2Part | undefined
+      this.showNextBorder = !tv2Part || !this.videoClipAction ? false : tv2Part?.metadata?.actionId === this.videoClipAction?.id
+    })
+
+    await this.rundownStateService.subscribeToRundown(this.segment.rundownId).then(rundownObservable => {
```

Comment:
Just an FYI: The current implementation of `subscribeToRundown` is firing a lot of events. Just keep that in mind when deciding to use this one :) 

Also: This doesn't clear the borders when we Reset the Rundown.